### DISCUSSION TITLE
`ECONNREFUSED` when `npm adduser`

### DISCUSSION THREAD

**Discussion by zhenyulin** [verdaccio/verdaccio#1410] [Q&A] [answered]
**Describe the bug**
Installed Verdaccio via the stable/verdaccio Helm Chart, added SSL via cert-manager. When doing `npm adduser --registry https://verdaccio.domain.com`, npm throws an error of 
```
npm ERR! code ECONNREFUSED
npm ERR! errno ECONNREFUSED
npm ERR! FetchError: request to https://verdaccio.domain.com/-/v1/login failed, reason: connect ECONNREFUSED IP_ADDRESS:443
```

got the same error code when using http instead of https

In verdaccio container log, there isn't the access log of those requests.

**Configuration File (cat ~/.config/verdaccio/config.yaml)**
https://github.com/helm/charts/blob/master/stable/verdaccio/values.yaml

**Environment information**

**Additional context**

* running K8S on GKE
* use nginx as Ingress
* npm@6.3.0
* `$ npm config get registry` prints: the right url of the registry ingress

**Comment by zhenyulin** [comment, marked as answer]
fixed this by running `npm config set always-auth true`

it would be nice to have this command in the initial UI homepage

**Comment by juanpicado** [comment]
Mmmmm, that might depends of the package manager you are using, `yarn` is enabled by default.
Related with https://github.com/verdaccio/verdaccio/issues/1381#issuecomment-511699858

I think rather `npm` should be more informative about when they do not are sending the token to the registry. This is a really common topic in the user side.

**Comment by zhenyulin** [comment]
seems getting the error code again for publishing a package or doing a login even when user creation is successful
